src/cargo/util/toml/mod.rs

```diff
@@ -644,6 +648,9 @@ impl TomlManifest {
         };
         let profiles = build_profiles(&me.profile);
         let publish = project.publish.unwrap_or(true);
+        let empty = Vec::new();
+        let cargo_features = project.cargo_features.as_ref().unwrap_or(&empty);
```

Comment:
Would `unwrap_or(&[])` work here?

Reply:
Oh that just required munging around with `Option<&Vec<T>>` -> `Option<&[T]>` which was just a pain.